recipe/meta.yaml

```diff
@@ -13,157 +14,236 @@
+  number: {{ build_num }}
+  skip: true  # [(py < 39) or cuda_compiler_version not in ("11.8", "12.0")]
 
 requirements:
-  build:
-    - {{ compiler("c") }}
-    - {{ compiler("cxx") }}
-    - {{ compiler("cuda") }}
-    #- cuda-nvcc
-    - sysroot_{{ target_platform }} 2.17    # [linux]
-    - cross-python_{{ target_platform }}    # [build_platform != target_platform]
-    - python                                # [build_platform != target_platform]
-    - cython >=0.29.22,<3                   # [build_platform != target_platform]
-    # TODO: clean up
-    {% if cuda_major >= 12 %}
-    - cuda-driver-dev                       # [linux and build_platform != target_platform]
-    - cuda-cudart-dev                       # [build_platform != target_platform]
-    - cuda-nvrtc-dev                        # [build_platform != target_platform]
-    - cuda-nvtx-dev                         # [build_platform != target_platform]
-    - cuda-profiler-api                     # [build_platform != target_platform]
-    - cuda-cccl_{{ build_platform }}        # [build_platform != target_platform]
-    - libcublas-dev                         # [build_platform != target_platform]
-    - libcufft-dev                          # [build_platform != target_platform]
-    - libcurand-dev                         # [build_platform != target_platform]
-    - libcusolver-dev                       # [build_platform != target_platform]
-    - libcusparse-dev                       # [build_platform != target_platform]
-    {% endif %}
-    # optional dependencies for CUDA 11.2+
-    - cudnn >=8.0.*,<9                      # [build_platform != target_platform and (not ((aarch64 or ppc64le) and (cuda_compiler_version or "").startswith("12")))]
-    - nccl >=2.8,<3                         # [build_platform != target_platform]
-    - cutensor >=1.4,<2                     # [build_platform != target_platform]
-    - cusparselt ~=0.2.0.0                  # [build_platform != target_platform and (linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
-
   host:
     - python
-    - pip
-    - setuptools
-    - cython >=0.29.22,<3
-    - fastrlock >=0.5
-    - cuda-version {{ cuda_compiler_version }}
-    - nvtx-c  # [win64]
-    # TODO: clean up
-    {% if cuda_major >= 12 %}
-    - cuda-driver-dev  # [linux]
-    - cuda-cudart-dev
-    - cuda-nvrtc-dev
-    - cuda-nvtx-dev
-    - cuda-profiler-api
-    - cuda-cccl_{{ target_platform }}
-    - libcublas-dev
-    - libcufft-dev
-    - libcurand-dev
-    - libcusolver-dev
-    - libcusparse-dev
-    {% endif %}
-    # optional dependencies
-    # TODO: see https://github.com/conda-forge/cudnn-feedstock/issues/58
-    - cudnn >=8.0.*,<9      # [not ((aarch64 or ppc64le) and (cuda_compiler_version or "").startswith("12"))]
-    - nccl >=2.8,<3         # [linux]
-    - cutensor >=1.4,<2
-    - cusparselt ~=0.2.0.0  # [(linux64 or aarch64 or win) and (cuda_compiler_version or "").startswith("11")]
```

Comment:
Think we may want to keep these top-level `requirements` at least for `build` & `host`

Reply:
Why? TBH I don't quite understand the purpose of top-level requirements, when there are multiple outputs. Seems like just hacks to me for bowing to the limitations of conda-build & co? But what do we wish to achieve here, when it's working fine now?

Reply:
The original idea of the multi-`outputs` recipe was to largely handle everything (like the build, dependencies at build time, etc.) at the top-level

The `outputs` sections themselves were mainly geared around taking the resulting contents of `$PREFIX` and slicing them up in different ways to create several (potentially interrelated) packages

Over time people have added more things to the subpackages sections (different requirement stages, more complicated build & install patterns, etc.). As we have seen these later additions are incomplete

This is why we see issues with Python and `PIP_*` environment variables, misunderstandings by the build tooling about how to build the recipe, etc.

Hewing close to the original model will minimize the chance of niche issues cropping up

Reply:
Thanks, John, for sharing the background. My preference is to keep the recipe as-is and see what else issue we could run into, unless you have strong opinion otherwise. The status quo is cleaner and makes better sense to reflect the intention IMHO. I am also very curious to see how this would involve into, once the new recipe format is up and running.